launcher/util.go

```diff
@@ -28,3 +31,20 @@ func FetchImpersonatedToken(ctx context.Context, serviceAccount string, audience
 
 	return []byte(token.AccessToken), nil
 }
+
+func listFilesWithPrefix(targetDir string, prefix string) ([]string, error) {
+	targetFiles := make([]string, 0)
+	err := filepath.WalkDir(targetDir, func(path string, d os.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
```

Comment:
The error handling logics should be carefully evaluated. From https://pkg.go.dev/io/fs#WalkDirFunc looks like simply returning a non-nil error will stop `WalkDir` walking the entire tree, is this the intended behavior?

Reply:
We will skip the directories for which `err` argument is set to non-nil during the `WalkDirFunc` call.